azurerm/internal/services/securitycenter/device_security_group_resource.go

```diff
@@ -0,0 +1,571 @@
+		Update: resourceDeviceSecurityGroupCreateUpdate,
+		Delete: resourceDeviceSecurityGroupDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.DeviceSecurityGroupID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"target_resource_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"allow_list_rule": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(security.RuleTypeConnectionToIPNotAllowed),
+								string(security.RuleTypeLocalUserNotAllowed),
+								string(security.RuleTypeProcessNotAllowed),
+							}, false),
+						},
+
+						"values": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+
+			"time_window_rule": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(security.RuleTypeActiveConnectionsNotInAllowedRange),
+								string(security.RuleTypeAmqpC2DMessagesNotInAllowedRange),
+								string(security.RuleTypeMqttC2DMessagesNotInAllowedRange),
+								string(security.RuleTypeHTTPC2DMessagesNotInAllowedRange),
+								string(security.RuleTypeAmqpC2DRejectedMessagesNotInAllowedRange),
+								string(security.RuleTypeMqttC2DRejectedMessagesNotInAllowedRange),
+								string(security.RuleTypeHTTPC2DRejectedMessagesNotInAllowedRange),
+								string(security.RuleTypeAmqpD2CMessagesNotInAllowedRange),
+								string(security.RuleTypeMqttD2CMessagesNotInAllowedRange),
+								string(security.RuleTypeHTTPD2CMessagesNotInAllowedRange),
+								string(security.RuleTypeDirectMethodInvokesNotInAllowedRange),
+								string(security.RuleTypeFailedLocalLoginsNotInAllowedRange),
+								string(security.RuleTypeFileUploadsNotInAllowedRange),
+								string(security.RuleTypeQueuePurgesNotInAllowedRange),
+								string(security.RuleTypeTwinUpdatesNotInAllowedRange),
+								string(security.RuleTypeUnauthorizedOperationsNotInAllowedRange),
+							}, false),
+						},
+
+						"max_threshold": {
```

Comment:
what is this the max of?

Reply:
Users will get alert when the data in the time window is not in the threashold [min, max]

Reply:
sounds like this is a range rule for a certain duration, so it would be clear if the schema was:
```
range_rule {
  type
  max
  min
  duration
}
```

i get what time winder is trying to say, but its rather.. unclear (in/outside this range for this duration) so the terms range, max, min, and duration make more sense?

Reply:
have updated